[Bug Report] Unable to use Rootless Docker due to hardcoded socket path

### Platform

Ubuntu 24.04.3 LTS

### Panel Version

10.11.0

### Daemon Version (Optional)

4.11.0

### Recurrence Probability

100%

### Bug Description

When using rootless Docker, the panel fails to pull or run images and shows:
```
[ERROR] Image download error, please make sure the image name is correct, or manually pull the required image using docker pull via the node management terminal, error:
connect ENOENT /var/run/docker.sock
```

In app.js (line 6240), the code appears to support the `DOCKER_HOST` env. variable:
```
socketPath: process.env.DOCKER_HOST ?? (os.platform() == "win32" ? "//./pipe/docker_engine" : "/var/run/docker.sock")
```
However, `DOCKER_HOST` is never actually used. Even when the variable is correctly set (e.g. `unix:///run/user/<UID>/docker.sock`), the daemon still connects to `/var/run/docker.sock`. As a result, rootless Docker setups fail because the socket path is effectively hard coded somehow.

**Expected behavior:**
The panel should respect the `DOCKER_HOST` environment variable, allowing users to point to any valid Docker socket, e.g. `/run/user/<UID>/docker.sock`.

**Steps to recreate:**
1. Install Docker
2. Disable rootfull Docker and remove its socket (located under `/var/run/docker.sock`)
3. Create a non-root user to run rootless Docker and MCSManager
4. Set up Docker to run in rootless mode for that user
5. Install MCSManager under the same user and make it run as that user
6. Ensure Docker and MCSManager run under the same non-root user
7. Try creating a new Docker instance from the MCSManager web panel

Interestingly, the error message itself appears to respect the `DOCKER_HOST` env. variable, which makes this issue even more confusing. For example, when `DOCKER_HOST` is set to `unix:///run/user/1001/docker.sock` the panel produces this error:
```
[ERROR] Image download error, please make sure the image name is correct, or manually pull the required image using docker pull via the node management terminal, error:
connect ENOENT unix:///run/user/1001/docker.sock
```
This suggests that `DOCKER_HOST` has been applied correctly, while in reality the daemon continues to use the hard coded socket path `/var/run/docker.sock`.

--- a/daemon/src/service/docker_service.ts
+++ b/daemon/src/service/docker_service.ts
@@ -2,15 +2,78 @@ import Docker from "dockerode";
 import { Readable } from "node:stream";
 import os from "os";
 
+function resolveDockerOptionsFromEnv(): Docker.DockerOptions {
+  const dockerHost = process.env.DOCKER_HOST?.trim();
+  if (dockerHost) {
+    const options = parseDockerHostToDockerOptions(dockerHost);
+    if (options) return options;
+  }
+  return {
+    socketPath: os.platform() == "win32" ? "//./pipe/docker_engine" : "/var/run/docker.sock"
+  };
+}
+
+function parseDockerHostToDockerOptions(dockerHost: string): Docker.DockerOptions | undefined {
+  // Support both:
+  // - Docker CLI style: unix:///run/user/<uid>/docker.sock (rootless Docker)
+  // - Raw socket path: /run/user/<uid>/docker.sock
+
+  const value = dockerHost.trim();
+  if (!value) return undefined;
+
+  // Windows named pipe (Docker CLI uses: npipe:////./pipe/docker_engine)
+  if (value.startsWith("npipe:")) {
+    let pipePath = value.slice("npipe:".length);
+    // Reduce leading slashes: "////./pipe/docker_engine" -> "//./pipe/docker_engine"
+    if (pipePath.startsWith("////")) pipePath = `//${pipePath.slice(4)}`;
+    // URL parsing would normalize away "./", keep the canonical dockerode form.
+    if (pipePath.startsWith("//pipe/")) pipePath = pipePath.replace("//pipe/", "//./pipe/");
+    return { socketPath: pipePath };
+  }
+
+  // Unix socket URL (Docker CLI style)
+  if (value.startsWith("unix://")) {
+    const socketPath = value.slice("unix://".length);
+    return socketPath ? { socketPath } : undefined;
+  }
+
+  // Support TCP/HTTP(S) style (e.g. tcp://127.0.0.1:2375)
+  if (value.includes("://")) {
+    try {
+      const url = new URL(value);
+
+      if (url.protocol === "unix:") {
+        return url.pathname ? { socketPath: url.pathname } : undefined;
+      }
+
+      if (url.protocol === "tcp:" || url.protocol === "http:" || url.protocol === "https:") {
+        const tlsEnabled = process.env.DOCKER_TLS_VERIFY?.trim() === "1";
+        const protocol =
+          url.protocol === "https:" ? "https" : tlsEnabled ? "https" : "http";
+        const defaultPort = protocol === "https" ? 2376 : 2375;
+        const port = url.port ? Number(url.port) : defaultPort;
+        return {
+          protocol,
+          host: url.hostname,
+          port
+        };
+      }
+    } catch {
+      // Fall through to raw socket path handling.
+    }
+  }
+
+  // Raw path fallback (common in systemd env files)
+  return { socketPath: value };
+}
+
 export class DefaultDocker extends Docker {
   public static readonly defaultConfig: Docker.DockerOptions = {
-    socketPath:
-      process.env.DOCKER_HOST ??
-      (os.platform() == "win32" ? "//./pipe/docker_engine" : "/var/run/docker.sock")
+    ...resolveDockerOptionsFromEnv()
   };
 
   constructor(p?: Docker.DockerOptions) {
-    super(Object.assign(p ?? {}, DefaultDocker.defaultConfig));
+    super(Object.assign({}, DefaultDocker.defaultConfig, p ?? {}));
   }
 }
 

